aws/resource_aws_neptune_cluster_parameter_group_test.go

```diff
@@ -0,0 +1,228 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSNeptuneClusterParameterGroupExists("aws_neptune_cluster_parameter_group.test", &v),
+					resource.TestMatchResourceAttr(
+						"aws_neptune_cluster_parameter_group.test", "name", regexp.MustCompile("^tf-test-")),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSNeptuneClusterParameterGroup_generatedName(t *testing.T) {
+	var v neptune.DBClusterParameterGroup
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSNeptuneClusterParameterGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSNeptuneClusterParameterGroupConfig_generatedName,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSNeptuneClusterParameterGroupExists("aws_neptune_cluster_parameter_group.test", &v),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSNeptuneClusterParameterGroup_withoutParameter(t *testing.T) {
+	var v neptune.DBClusterParameterGroup
+
+	parameterGroupName := fmt.Sprintf("cluster-parameter-group-test-tf-%d", acctest.RandInt())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSNeptuneClusterParameterGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSNeptuneClusterParameterGroupOnlyConfig(parameterGroupName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSNeptuneClusterParameterGroupExists("aws_neptune_cluster_parameter_group.bar", &v),
+					testAccCheckAWSNeptuneClusterParameterGroupAttributes(&v, parameterGroupName),
+					resource.TestCheckResourceAttr(
+						"aws_neptune_cluster_parameter_group.bar", "name", parameterGroupName),
+					resource.TestCheckResourceAttr(
+						"aws_neptune_cluster_parameter_group.bar", "family", "neptune1"),
+					resource.TestCheckResourceAttr(
+						"aws_neptune_cluster_parameter_group.bar", "description", "Managed by Terraform"),
+					resource.TestCheckResourceAttr("aws_neptune_cluster_parameter_group.bar", "parameter.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSNeptuneClusterParameterGroupDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).neptuneconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_neptune_cluster_parameter_group" {
+			continue
+		}
+
+		resp, err := conn.DescribeDBClusterParameterGroups(
+			&neptune.DescribeDBClusterParameterGroupsInput{
+				DBClusterParameterGroupName: aws.String(rs.Primary.ID),
+			})
+
+		if err == nil {
+			if len(resp.DBClusterParameterGroups) != 0 &&
+				aws.StringValue(resp.DBClusterParameterGroups[0].DBClusterParameterGroupName) == rs.Primary.ID {
+				return errors.New("Neptune Cluster Parameter Group still exists")
+			}
+		}
+
+		if err != nil {
+			if isAWSErr(err, neptune.ErrCodeDBParameterGroupNotFoundFault, "") {
```

Comment:
The SDK provides a separate constant for cluster parameter groups: `ErrCodeDBClusterParameterGroupNotFoundFault`

Reply:
@bflad Looks like `ErrCodeDBClusterParameterGroupNotFoundFault ` only available for createdbcluster and modifydbcluster and `ErrCodeDBParameterGroupNotFoundFault ` is the one used for other functions.
https://docs.aws.amazon.com/sdk-for-go/api/service/neptune/#Neptune.DeleteDBClusterParameterGroup

Reply:
🙁 this is either an error in the AWS documentation or an oversight. I will reach out to AWS support for clarification.

Reply:
Sorry this was a bad assumption on my part. This behavior is frustrating:

```
$ aws neptune describe-db-cluster-parameter-groups --db-cluster-parameter-group-name does-not-exist

An error occurred (DBParameterGroupNotFound) when calling the DescribeDBClusterParameterGroups operation: DBClusterParameterGroup not found: does-not-exist

$ aws neptune delete-db-cluster-parameter-group --db-cluster-parameter-group-name does-not-exist

An error occurred (DBParameterGroupNotFound) when calling the DeleteDBClusterParameterGroup operation: DBClusterParameterGroup not found: does-not-exist

$ aws neptune modify-db-cluster-parameter-group --db-cluster-parameter-group-name does-not-exist --parameters 'ParameterName=test,ParameterValue=""'

An error occurred (DBParameterGroupNotFound) when calling the ModifyDBClusterParameterGroup operation: Parameter group does not exist
```

I guess we should stick with the non-`Cluster` constant and add a comment to each instance about why its not using the `Cluster` constant. My apologies.

Reply:
yes, acceptance tests were failed when I tried `ErrCodeDBClusterParameterGroupNotFoundFault `.